xbmc/platform/android/jni/AudioFormat.cpp

```diff
@@ -84,6 +86,10 @@ void CJNIAudioFormat::PopulateStaticFields()
         {
           CJNIAudioFormat::ENCODING_DTS = get_static_field<int>(c, "ENCODING_DTS");
           CJNIAudioFormat::ENCODING_DTS_HD = get_static_field<int>(c, "ENCODING_DTS_HD");
+          // Nvidia Shield v6 firmware uses another ID, which will also be the future ID
+          // though other v23 version would not return a value if we'd use the get_static_field
+          // method to query this value. The hardcoded value can be removed after probably is out
+          CJNIAudioFormat::ENCODING_DOLBY_TRUEHD = 0x0000000d;
```

Comment:
I thought about letting it run into an exception, but as this value does
not exist "officially" in v22 nor v23 I decided for this way. When v24 is
released. Also as Nvidia did not document that value anywhere - I did not
know that it returns a proper value even on the shield, which was another
reason I decided for hard coding.

Thanks much.

2016-02-03 9:51 GMT+01:00 Chris Browet <notifications@github.com>:

> In xbmc/platform/android/jni/AudioFormat.cpp
> <https://github.com/xbmc/xbmc/pull/9023#discussion_r51693019>:
>
> > @@ -84,6 +86,10 @@ void CJNIAudioFormat::PopulateStaticFields()
> >          {
> >            CJNIAudioFormat::ENCODING_DTS = get_static_field<int>(c, "ENCODING_DTS");
> >            CJNIAudioFormat::ENCODING_DTS_HD = get_static_field<int>(c, "ENCODING_DTS_HD");
> > +          // Nvidia Shield v6 firmware uses another ID, which will also be the future ID
> > +          // though other v23 version would not return a value if we'd use the get_static_field
> > +          // method to query this value. The hardcoded value can be removed after probably is out
> > +          CJNIAudioFormat::ENCODING_DOLBY_TRUEHD = 0x0000000d;
>
> See
> https://github.com/koying/xbmc/blob/master/xbmc/android/jni/AudioFormat.cpp#L97
> and
> https://github.com/koying/xbmc/blob/master/xbmc/android/jni/AudioFormat.cpp#L57
> to get cross-api values while taking into account the fact that the value
> might not be there
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/9023/files#r51693019>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B